### ISSUE TITLE
[yeelight] Autodiscovery is not working

### ISSUE THREAD

**Issue by vkoop** [openhab/openhab-addons#6643]
## Expected Behavior

Yeelight devices should be detected automatically.

## Current Behavior

The binding tries to autodetect using all network interfaces even loopback interfaces and inactive interfaces this seems to lead to the problem that the yeelights aren't discovered at all.

## Possible Solution

The DeviceManager class could be improved to use only active network interfaces. Furthermore it is enough to trigger the discovery broadcast once.

## Steps to Reproduce (for Bugs)

I tried to find the yeelight using PaperUI.

## Your Environment

OS: Manjaro Linux / Windows 10 both didn't work
Multiple network interfaces.

**Comment by Hilbrand** [comment]
Closed by #6644

**Comment by claell** [comment]
@vkoop thanks for your work on the Yeelight binding! Just for my understanding, in this issue and the related PR you partly revert the changes introduced in #4573 and limit discovery to only active interfaces? This means not completely reverting #4573, but improving the solution to better chose which interfaces to use?

**Comment by vkoop** [comment]
@claell In my previous merge request #6644 i've limited the network devices to active interfaces.

In the merge request #6644 there was also the remark that it isn't optimal that on every run a new a new fixed thread pool was created. I've changed the thread pool to a cached thread pool. 

Furthermore I've moved the logic for the lookup of the network interfaces to a method in preparation of writing some unit tests (it would be easier to mock the method call).
